malariagen_data/util.py

```diff
@@ -839,3 +884,43 @@ def check_colab_location(*, gcs_url: str, url: str) -> Optional[ipinfo.details.D
             pass
 
     return details
+
+
+def check_types(f):
```

Comment:
N.B., we roll our own simple decorator here for performing runtime type checks on parameter values. See the docstring for explanation of why we don't use typeguard's `@typechecked` decorator.